site/schemas/create-disruption.schema.ts

```diff
@@ -190,6 +201,87 @@ export const createDisruptionsSchemaRefined = createDisruptionSchema
             path: ["disruptionStartDate"],
             message: "Validity periods cannot overlap",
         },
-    );
+    )
+    .refine(
```

Comment:
We might be missing a couple of checks on the top level validity period fields, we have them for the validity array but not the top level `disruptionStartDate` etc.
```ts
.refine((val) => (val.disruptionEndDate ? !!val.disruptionEndTime : true), {
    path: ["disruptionEndTime"],
    message: "Disruption end time must be set when end date is set",
})
.refine((val) => (val.disruptionEndTime ? !!val.disruptionEndDate : true), {
    path: ["disruptionEndDate"],
    message: "Disruption end date must be set when end time is set",
})
.refine(
    (val) => {
        if (val.disruptionEndDate && val.disruptionEndTime) {
            return getDatetimeFromDateAndTime(val.disruptionEndDate, val.disruptionEndTime).isAfter(
                getDatetimeFromDateAndTime(val.disruptionStartDate, val.disruptionStartTime),
            );
        }

        return true;
    },
    {
        path: ["disruptionEndDate"],
        message: "Disruption end datetime must be after start datetime",
    },
)
```

Reply:
Although it might be better if we reuse the validity schema instead we could update the `createDisruptionSchema` to be:
```ts
export const createDisruptionSchema = z
  .object({
    disruptionType: z.union(
        [z.literal("planned"), z.literal("unplanned")],
        setZodDefaultError("Select a disruption type"),
    ),
    summary: z.string(setZodDefaultError("Enter a summary for this disruption")).min(1).max(100, {
        message: "Summary must not exceed 100 characters",
    }),
    description: z.string(setZodDefaultError("Enter a description for this disruption")).min(1).max(500, {
        message: "Description must not exceed 500 characters",
    }),
    associatedLink: z
        .string()
        .url({
            message: "Associated link must be a valid URL",
        })
        .max(500, {
            message: "Associated link must not exceed 500 characters",
        })
        .optional()
        .or(z.literal("")),
    disruptionReason: z.union(
        [miscellaneousReasonSchema, environmentReasonSchema, personnelReasonSchema, equipmentReasonSchema],
        setZodDefaultError("Select a reason from the dropdown"),
    ),
    publishStartDate: zodDate("Enter a publish start date for the disruption"),
    publishStartTime: zodTime("Enter a publish start time for the disruption"),
    publishEndDate: zodDate("Invalid publish end date").optional().or(z.literal("")),
    publishEndTime: zodTime("Invalid publish end date").optional().or(z.literal("")),
    validity: z
        .array(validitySchemaRefined)
        .refine((arr) => !arr.some((val) => val.disruptionNoEndDateTime === "true"), {
            path: ["disruptionStartDate"],
            message: "A validity period with no end time must be the last validity",
        })
        .optional(),
})
.and(validitySchemaRefined);
```

Reply:
`schema.shape` was not recognised using intersection type. Have use the block of code above the intersection type code